src/main/kotlin/jp/s6n/idea/rustowl/configuration/AppSettingsComponent.kt

```diff
@@ -0,0 +1,254 @@
+    }
+
+    private fun configureInstallButton() {
+        constraints.gridx = 0
+        constraints.gridy = 2
+        mainPanel.add(JBLabel("Install cargo-owlsp:"), constraints)
+        constraints.gridx = 1
+        installButton.addActionListener {
+            installCargoOwlsp()
+        }
+        mainPanel.add(installButton, constraints)
+    }
+
+    private fun configureColorPickers() {
+        DecorationType.entries.forEachIndexed { index, type ->
+            constraints.gridx = 0
+            constraints.gridy = index + 1
+            mainPanel.add(JLabel(type.displayText), constraints) // Use displayText
+
+            val button = JButton().apply {
+                background = type.defaultColor
+                addActionListener { chooseColor(this, type) }
+            }
+            colorChoosers[type] = button
+
+            constraints.gridx = 1
+            mainPanel.add(button, constraints)
+        }
+    }
+
+    private fun chooseColor(button: JButton, type: DecorationType) {
+        val newColor = JColorChooser.showDialog(null, "Choose color for ${type.displayText}", button.background)
+        if (newColor != null) {
+            button.background = newColor
+        }
+    }
+
+    private fun checkCargoOwlspInPath() {
+        val command = if (System.getProperty("os.name").startsWith("Windows")) {
+            "where cargo-owlsp"
+        } else {
+            "which cargo-owlsp"
+        }
+
+        try {
+            val process = ProcessBuilder(command.split(" "))
+                .redirectErrorStream(true)
+                .start()
+            val exitCode = process.waitFor()
+            if (exitCode == 0) {
+                val path = process.inputStream.bufferedReader().readLine()
+                cargoOwlspPathField.text = path
+                installButton.isEnabled = false
+            } else {
+                cargoOwlspPathField.text = "Not found"
+                installButton.isEnabled = true
+            }
+        } catch (e: IOException) {
+            cargoOwlspPathField.text = "Error checking PATH"
+            installButton.isEnabled = true
+        }
+    }
+
+    private fun getCargoOwlspPath(): String? {
+        val command = if (System.getProperty("os.name").startsWith("Windows")) {
+            "where cargo-owlsp"
+        } else {
+            "which cargo-owlsp"
+        }
+
+        return try {
+            val process = ProcessBuilder(command.split(" "))
+                .redirectErrorStream(true)
+                .start()
+
+            if (process.waitFor() == 0) {
+                process.inputStream.bufferedReader().readLine()
+            } else {
+                logger.error("Failed to get path from input stream")
+                null
+            }
+        } catch (e: IOException) {
+            logger.error("Failed to get path from input stream", e)
+            null
+        }
+    }
+
+    private fun installCargoOwlsp() {
```

Comment:
IMO installing RustOwl automatically is too out-of-scope for this plugin, we can consider adding this or not later. At least the official extension for VS Code doesn't have this feature at this moment.